http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmApiClientProvider.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import bio.terra.workspace.api.ControlledAzureResourceApi
+import bio.terra.workspace.client.ApiClient
+import org.http4s.Uri
+
+/**
+ * Represents a way to get a client for interacting with workspace manager controlled resources.
+ * Additional WSM clients can be added here if needed.
+ *
+ * Based on `org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/WorkspaceManagerApiClientProvider.scala`
+ *
+ */
+trait WsmApiClientProvider {
```

Comment:
I actually copied this from Lucy's [PR](https://github.com/DataBiosphere/leonardo/pull/3401) because I wanted to use the Java client. :) Whoever merges second will need to rebase, but the code is the same.

Reply:
@lucymcnatt just FYI